modules/tinymce/src/core/test/ts/browser/html/StylesTest.ts

```diff
@@ -234,4 +237,124 @@ describe('browser.tinymce.core.html.StylesTest', () => {
     assertStyles(styles, 'background:url(javascript:alert(1)', `background: url('javascript:alert(1');`);
     assertStyles(styles, 'background:url(vbscript:alert(1)', `background: url('vbscript:alert(1');`);
   });
+
+  const applyAndAssertForeColor = (editor: Editor, appliedColor: string, expectedColor: string) => {
+    // Reset content
+    editor.setContent('<p>colour me</p>');
+
+    return {
+      /** Apply color using 'mceApplyTextcolor' command. */
+      usingCommand: () => {
+        TinySelections.setSelection(editor, [ 0, 0 ], 0, [ 0, 0 ], 'color me'.length);
+        editor.execCommand('mceApplyTextcolor', 'forecolor' as any, appliedColor);
+        TinyAssertions.assertContentPresence(editor, { [`span[data-mce-style="color: ${expectedColor};"]`]: 1 });
+      },
+      /** Apply color using the 'forecolor' part of the toolbar. */
+      usingToolbar: async () => {
+        TinyUiActions.clickOnToolbar(editor, '[aria-label^="Text color"] > .tox-tbtn + .tox-split-button__chevron');
+        await TinyUiActions.pWaitForUi(editor, '.tox-swatches');
+        TinyUiActions.clickOnUi(editor, `div[data-mce-color="${appliedColor}"]`);
+        TinyAssertions.assertContentPresence(editor, { [`span[data-mce-style="color: ${expectedColor};"]`]: 1 });
```

Comment:
Maybe assert both the presence that is basically the current view state and the content using assertContent that is the serialized output state and that should be the expected forced color as well.

Reply:
Added `assertContent` calls as well